File: sdk/android/plugin-network/src/main/kotlin/dev/probe/network/HttpTransaction.kt
package dev.probe.network

data class HttpTransaction(
    val id: String,
    val timestamp: Long,
    val method: String,
    val url: String,
    val requestHeaders: Map<String, String>,
    val requestBody: String?,
    val requestSizeBytes: Long,
    val responseCode: Int?,
    val responseMessage: String?,
    val responseHeaders: Map<String, String>,
    val responseBody: String?,
    val responseSizeBytes: Long?,
    val durationMs: Long?,
    val error: String?
)

internal fun HttpTransaction.toPayload(): Map<String, Any?> = mapOf(
    "id" to id,
    "timestamp" to timestamp,
    "method" to method,
    "url" to url,
    "requestHeaders" to requestHeaders,
    "requestBody" to requestBody,
    "requestSizeBytes" to requestSizeBytes,
    "responseCode" to responseCode,
    "responseMessage" to responseMessage,
    "responseHeaders" to responseHeaders,
    "responseBody" to responseBody,
    "responseSizeBytes" to responseSizeBytes,
    "durationMs" to durationMs,
    "error" to error
)

File: sdk/ios/Sources/ProbeNetwork/HttpTransaction.swift
import Foundation

/// Represents a captured HTTP request/response pair.
public struct HttpTransaction {
    public let id: String
    public let url: String
    public let method: String
    public let requestHeaders: [String: String]
    public let requestBody: String?
    public let statusCode: Int?
    public let responseHeaders: [String: String]
    public let responseBody: String?
    public let durationMs: Double
    public let requestSize: Int
    public let responseSize: Int
    public let timestamp: Double

    public init(
        id: String = UUID().uuidString,
        url: String,
        method: String,
        requestHeaders: [String: String] = [:],
        requestBody: String? = nil,
        statusCode: Int? = nil,
        responseHeaders: [String: String] = [:],
        responseBody: String? = nil,
        durationMs: Double = 0,
        requestSize: Int = 0,
        responseSize: Int = 0,
        timestamp: Double = Date().timeIntervalSince1970 * 1000
    ) {
        self.id = id; self.url = url; self.method = method
        self.requestHeaders = requestHeaders; self.requestBody = requestBody
        self.statusCode = statusCode; self.responseHeaders = responseHeaders
        self.responseBody = responseBody; self.durationMs = durationMs
        self.requestSize = requestSize; self.responseSize = responseSize
        self.timestamp = timestamp
    }

    func toPayload() -> [String: Any?] {
        [
            "id": id, "url": url, "method": method,
            "requestSize": requestSize, "responseSize": responseSize,
            "statusCode": statusCode, "durationMs": durationMs,
            "timestamp": timestamp,
        ]
    }
}
